linera-rpc/src/grpc/client.rs

```diff
@@ -316,6 +316,34 @@ impl ValidatorNode for GrpcClient {
         Ok(client_delegate!(self, download_certificate, hash)?.try_into()?)
     }
 
+    #[instrument(target = "grpc_client", skip_all, err, fields(address = self.address))]
+    async fn download_certificates(
+        &self,
+        hashes: Vec<CryptoHash>,
+    ) -> Result<Vec<Certificate>, NodeError> {
+        let mut missing_hashes = hashes;
+        let mut certs_collected = Vec::with_capacity(missing_hashes.len());
+        loop {
+            // Macro doesn't compile if we pass `missing_hashes.clone()` directly to `client_delegate!`.
```

Comment:
(I guess we could evaluate the expression at the beginning of the macro, and assign it to a local variable.)